maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -123,115 +120,109 @@ public fun GoogleMap(
 
     val parentComposition = rememberCompositionContext()
     val currentContent by rememberUpdatedState(content)
-    LaunchedEffect(Unit) {
-        disposingComposition {
-            mapView.newComposition(parentComposition, mapClickListeners) {
-                MapUpdater(
-                    mergeDescendants = mergeDescendants,
-                    contentDescription = currentContentDescription,
-                    cameraPositionState = currentCameraPositionState,
-                    contentPadding = currentContentPadding,
-                    locationSource = currentLocationSource,
-                    mapProperties = currentMapProperties,
-                    mapUiSettings = currentUiSettings,
-                )
-
-                MapClickListenerUpdater()
-
-                CompositionLocalProvider(
-                    LocalCameraPositionState provides currentCameraPositionState,
-                ) {
-                    currentContent?.invoke()
-                }
-            }
-        }
-    }
-}
 
-internal suspend inline fun disposingComposition(factory: () -> Composition) {
-    val composition = factory()
-    try {
-        awaitCancellation()
-    } finally {
-        composition.dispose()
-    }
-}
+    /**
+     * Create and apply the [content] compositions to the map +
+     * dispose the [Composition] when the parent composable is disposed.
+     * */
+    fun CoroutineScope.launchSubcomposition(mapView: MapView): Job {
+        val mapCompositionContent: @Composable () -> Unit = {
+            MapUpdater(
+                mergeDescendants = mergeDescendants,
+                contentDescription = currentContentDescription,
+                cameraPositionState = currentCameraPositionState,
+                contentPadding = currentContentPadding,
+                locationSource = currentLocationSource,
+                mapProperties = currentMapProperties,
+                mapUiSettings = currentUiSettings,
+            )
```

Comment:
It would be useful to move `launchSubcomposition()` to the top level, to help avoid direct references to `GoogleMap()` parameters from the long-running lambda closure. There have been subtle bugs in this project caused by such references, they're easy to miss.

This should incorporate the changes from recently approved #522, grouping together all `MapUpdater()` parameters.

Reply:
Alright 👍

Reply:
I've incorporated the changes from #522. Please tell me if there's anything you'd like to change.

Reply:
Looks great, thank you! The `MapUpdater()` changes are excellent.